docs/application/native/guides/text-input/voice-control-engine.md

```diff
@@ -0,0 +1,527 @@
+
+- Send the current state using the `vce_send_error()` function and send the error to a engine service user.
+
+  ```
+    void
+    send_error(void)
+    {
+        int ret = VCE_ERROR_NONE;
+
+        ret = vce_send_error(VCE_ERROR_NONE, "msg", NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Send the current state using the `vce_set_private_data()` function. Set private data to a engine service user.
+
+  ```
+    void
+    set_private_data(void)
+    {
+        int ret = VCE_ERROR_NONE;
+
+        ret = vce_set_private_data("key", "data");
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+
+<a name="info_retrieval"></a>
+## VCE Information Retrieval
+
+You can get the following information about the VCE:
+
+- Get all commands using the `vce_get_foreach_command()` function.
+
+  ```
+    bool
+    command_cb(int id, int type, int format, const char* command, const char* param, int domain, void* user_data)
+    {
+        /* Callback command data */
+        /* Your code */
+    }
+
+    void
+    get_foreach_command(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        int g_vce_command;
+
+        ret = vce_get_foreach_command(g_vce_command, command_cb, NULL);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Get command length using the `vce_get_command_count()` function. Get `g_vce_command` data from set_commands() function.
+
+  ```
+    void
+    get_command_count(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        int count = 0;
+
+        ret = vce_get_command_count(g_vce_command, &count);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+    }
+  ```
+
+- Get current audio type using the `vce_get_audio_type()` function.
+
+  ```
+    void
+    get_audio_type(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        char *audio_type = NULL;
+
+        ret = vce_get_audio_type(&audio_type);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+        if(NULL != audio_type) {
+            /* Memory release */
+            free(audio_type);
+            audio_type = NULL;
+        }
+    }
+  ```
+
+- Get private data from a voice manager client using the `vce_get_private_data()` function.
+
+  ```
+    void
+    get_private_data(void)
+    {
+        int ret = VCE_ERROR_NONE;
+        char *data = NULL;
+
+        ret = vce_get_private_data("key", &data);
+        if(VCE_ERROR_NONE != ret)
+            /* Error handling */
+        if(NULL != data) {
+            /* Memory release */
+            free(data);
+            data = NULL;
+        }
+    }
+  ```
+
+
+<a name="control_record"></a>
+## VCE Control Recording
+
+You can control recording of the following information about the VCE:
+
+- Start recording voice using the `vce_start_recording()` function. If the function call is successful, You can receive recording data from `set_recording` function.
```

Comment:
You can receive recording data from `set_recording` function. 

Please change:

you can receive recording data from `set_recording` function.

Reply:
Done.